src/CSharp/CodeCracker/Usage/SimplifyRedundantBooleanComparisonsAnalyzer.cs

```diff
@@ -31,18 +31,29 @@ public override void Initialize(AnalysisContext context)
             context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.EqualsExpression, SyntaxKind.NotEqualsExpression);
         }
 
-        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        private static void AnalyzeNode(SyntaxNodeAnalysisContext context)
         {
             var comparison = (BinaryExpressionSyntax)context.Node;
-            var semanticModel = context.SemanticModel;
 
-            var right = context.SemanticModel.GetConstantValue(comparison.Right);
-            if (!right.HasValue) return;
+            // Only handle the case where both operands of type bool; other cases involve
+            // too much complexity to be able to deliver an accurate diagnostic confidently.
+            var leftType = context.SemanticModel.GetTypeInfo(comparison.Left).Type;
+            var rightType = context.SemanticModel.GetTypeInfo(comparison.Right).Type;
```

Comment:
I don't know how it's possible... If the type is unknown, `TypeInfo.Type` returns a `IErrorTypeSymbol`, not null. But perhaps it's `GetTypeInfo` that returns null... I'll investigate